### PR TITLE
show the topics

### PR DESCRIPTION
**Pull request by leherreras**
As a user I should be able to see all the topics in the system

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,6 +1,11 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
+  namespace :api, defaults: { format: 'json' } do
+    namespace :v1 do
+      resources :topics
```
Comment (santir489) [inline config/routes.rb]: `resources :topics` generates routes for index, show, destroy, etc... only index is implemented
You can specify that!

File: spec/models/topic_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# Indexes
+#
+#  index_topics_on_name  (name) UNIQUE
+#
+require 'rails_helper'
+
+RSpec.describe Topic, type: :model do
+  let(:topic) { build(:topic) }
+
+  it 'validates name presence' do
```
Comment (santir489) [inline spec/models/topic_spec.rb]: Maybe we can use https://github.com/thoughtbot/shoulda-matchers#activerecord-matchers 
to validate name uniqueness and presence

File: spec/requests/api/v1/topic/topics_spec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe '/topics', type: :request do
+  subject { get api_v1_topics_url }
+  let!(:topics) { create_list(:topic, 3) }
+
+  describe 'GET /index' do
+    it 'how many topics' do
```
Comment (santir489) [inline spec/requests/api/v1/topic/topics_spec.rb]: `it how many topics`  sounds wierd right? The `it` description should be `it does something`
Maybe we can use `it 'returns all topcis' do`

Comment (santir489) [reply spec/requests/api/v1/topic/topics_spec.rb]: Also, are we testing that response has all attributes we want ?

**Review by santir489 — COMMENTED** [review]
Nice! Left some comments!

File: spec/requests/api/v1/topic/topics_spec.rb (hunk 3 above)
Comment (leherreras) [reply spec/requests/api/v1/topic/topics_spec.rb]: add the description 
`it 'renders a successful response' do`

File: config/routes.rb (hunk 1 above)
Comment (leherreras) [reply config/routes.rb]: add in routes
`resources :topics, only: :index`

File: spec/models/topic_spec.rb (hunk 2 above)
Comment (leherreras) [reply spec/models/topic_spec.rb]: now I am use the Shoulda gem

**Review by santir489 — APPROVED** [review]
👏

File: app/controllers/api/v1/topics_controller.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class TopicsController < ApplicationController
+      # before_action :set_topic, only: %i[show edit update destroy]
```
Comment (sebastiancaraballo) [inline app/controllers/api/v1/topics_controller.rb]: Avoid comments

**Review by sebastiancaraballo — APPROVED** [review]

File: app/controllers/api/v1/topics_controller.rb (hunk 4 above)
Comment (leherreras) [reply app/controllers/api/v1/topics_controller.rb]: thanks

**Pull request merged**